packages/okidoc-md/src/buildDocumentationSource/buildDocumentationSourceAST/buildDocumentationSourceAST.js

```diff
@@ -24,38 +25,66 @@ function buildDocumentationSourceAST({
     createApiFunction,
   } = getVisitorApi({ tag, visitorPath });
 
-  const interfaces = [];
-  const classDeclarations = [];
-  const classBodyItems = [];
-  const functions = [];
-
   // NOTE: read about visitors
   // https://github.com/thejameskyle/babel-handbook/blob/master/translations/en/plugin-handbook.md#toc-visitors
   // https://github.com/babel/babel/blob/master/packages/babel-traverse/src/visitors.js
-  const visitors = {
-    'InterfaceDeclaration|TSInterfaceDeclaration': (path, state) => {
-      const interfaceDeclaration = createApiInterface(path.node, path);
+  const interfaceVisitor = {
+    'GenericTypeAnnotation|TSTypeReference': (path, state) => {
+      const file = state.dependenciesTree.get(state.filePath);
+
+      if (file) {
+        // flow support with GenericTypeAnnotation, how to verify?
+        const typeName = t.isTSTypeReference(path.node) ? path.node.typeName.name : path.node.id.name;
+        // NOTE: it might be the case when more than 1 import exists.
+        // Example:
+        // import { MyType } from './types';
+        // import { MyType } from './another-type';
+        // in this case both of them would be added to the doc
+        file.dependencies.forEach((importSpecifiers, absoluteFilePath) => {
+          const importSpecifier = importSpecifiers.find(specifier => specifier.local.name === typeName);
+
+          if (importSpecifier) {
+            const interfaceFile = state.dependenciesTree.get(absoluteFilePath);
+
+            if (interfaceFile) {
+              const importName = t.isImportDefaultSpecifier(importSpecifier) ?
```

Comment:
Looks like this logic could be moved to the place where specifiers are extracted.
Not sure if we need some other info here except specifier name in this file.

Reply:
I had an idea in mind that `buildDependenciesTree` is responsible for building the tree itself, without knowing which content might be used in future for markdown generation process. 

But maybe it's possible to put interfaceVisitor logic inside createApiClassMethod / createApiClassProperty / createApiFunction methods?